HW_2/task_4.py

```diff
@@ -0,0 +1,40 @@
+"""
+Задайте список из N элементов, заполненный целыми числами из промежутка [-N, N].
+Найдите произведение элементов на индексах, хранящихся в файле indexes.txt (в одной строке один индекс).
+Решение должно работать при любом натуральном N.
+
+Ввод: значение типа <int>
+Вывод: значение типа <int>
+"""
+
+import random
+num = int(input('Введите число: '))
+new_list = [random.randint(-num, num) for r in range(num)]
+index_list = []
+prod = 1 
+with open('indexes.txt', encoding='UTF-8') as data:
+    for index in data:
+        index_list.append(int(index))
+for i in range(-num, num):
+    if i in index_list:
+        prod *= new_list[i]
+print(prod)
+
+
+import pathlib
+from pathlib import Path
+path = Path("indexes.txt") 
+
+
+number = int(input('Введите натуральное число: '))
+my_list = [Path(-number, number) for _ in range(number)]
```

Comment:
Ошибка у вас возникает здесь? Конструктор класса `Path` принимает только объекты типа `str`, а вы ему передаёте объекты класса `int`. Но мне не совсем понятна ваша логика здесь, ведь даже если бы у вас были объекты типа `str`, вы бы создали список из строковых элементов, причем одинаковых. Наверное, здесь нужно было все таки использовать функцию рэнд.
```suggestion
my_list = [random.randint(-num, num) for _ in range(number)]
```